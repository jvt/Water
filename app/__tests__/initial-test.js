import {
    Image,
    Text
} from 'react-native';
import React from 'react';
import Initial from '../initial';
import Button from 'apsl-react-native-button';
import Renderer from 'react-test-renderer';

describe('Initial', () => {
	test('renders correctly', () => {
		const screen = Renderer.create(
		<Initial />
		).toJSON();
		expect(screen).toMatchSnapshot();
	});

	test('WaterImage', () => {
		const instance = Renderer.create(
			<Image
			Style={{}}
			Droplet
			/>
		);

		expect(instance.toJSON()).toMatchSnapshot();
	})

	test('TitleText', () => {
		const instance  = Renderer.create(
			<Text
			style={{}}>
			Clean Water
			</Text>
		);

		expect(instance.toJSON()).toMatchSnapshot();
	});

	test('GroupText', () => {
		const instance  = Renderer.create(
			<Text
			style={{}}>
			sudo Give Us A's
			</Text>
		);

		expect(instance.toJSON()).toMatchSnapshot();
	});

	test('LoginButton', () => {
		const instance  = Renderer.create(
			<Button 
			style={{}}>
			Login
			</Button>
		);

		expect(instance.toJSON()).toMatchSnapshot();
	});

	test('RegisterButton', () => {
		const instance  = Renderer.create(
			<Button 
			style={{}}>
			Register
			</Button>
		);

		expect(instance.toJSON()).toMatchSnapshot();
	});

})