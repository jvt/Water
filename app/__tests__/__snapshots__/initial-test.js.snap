exports[`Initial GroupText 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={Object {}}>
  sudo Give Us A\'s
</Text>
`;

exports[`Initial LoginButton 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "borderRadius": 8,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 44,
      "justifyContent": "center",
      "marginBottom": 10,
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
          "fontSize": 18,
          "textAlign": "center",
        },
        undefined,
      ]
    }>
    Login
  </Text>
</View>
`;

exports[`Initial RegisterButton 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "borderRadius": 8,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 44,
      "justifyContent": "center",
      "marginBottom": 10,
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
          "fontSize": 18,
          "textAlign": "center",
        },
        undefined,
      ]
    }>
    Register
  </Text>
</View>
`;

exports[`Initial TitleText 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={Object {}}>
  Clean Water
</Text>
`;

exports[`Initial WaterImage 1`] = `
<Image
  Droplet={true}
  Style={Object {}} />
`;

exports[`Initial renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
          }
        }>
        <Image
          source={1}
          style={
            Object {
              "height": 175,
              "width": 80,
            }
          } />
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "200",
            "textAlign": "center",
          }
        }>
        Clean Water
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 15,
            "fontWeight": "200",
            "textAlign": "center",
          }
        }>
        sudo Give Us A\'s
      </Text>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "backgroundColor": "rgba(65, 163, 221, 1)",
            "borderRadius": 8,
            "borderWidth": 0,
            "flexDirection": "row",
            "height": 44,
            "justifyContent": "center",
            "marginBottom": 10,
            "marginLeft": 20,
            "marginRight": 20,
            "marginTop": 50,
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "flex": 1,
                "fontSize": 18,
                "textAlign": "center",
              },
              Object {
                "fontSize": 18,
              },
            ]
          }>
          Login
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "backgroundColor": "rgba(63, 209, 127, 1)",
            "borderRadius": 8,
            "borderWidth": 0,
            "flexDirection": "row",
            "height": 44,
            "justifyContent": "center",
            "marginBottom": 10,
            "marginLeft": 20,
            "marginRight": 20,
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "flex": 1,
                "fontSize": 18,
                "textAlign": "center",
              },
              Object {
                "fontSize": 18,
              },
            ]
          }>
          Register
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
